docs/01-app/02-guides/upgrading/version-16.mdx

```diff
@@ -0,0 +1,1072 @@
+
+## Turbopack by default
+
+Starting with **Next.js 16**, Turbopack is stable and used by default with `next dev` and `next build`
+
+Previously you had to enable Turbopack using `--turbopack`, or `--turbo`.
+
+```json filename="package.json"
+{
+  "scripts": {
+    "dev": "next dev --turbopack",
+    "build": "next build --turbopack",
+    "start": "next start"
+  }
+}
+```
+
+This is no longer necessary. You can update your `package.json` scripts:
+
+```json filename="package.json"
+{
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  }
+}
+```
+
+If you run `next build` without the `--turbopack` flag (the new default), and your Next config contains a `webpack` configuration, the **build will fail** to prevent misconfiguration issues. In this case, you must either:
+
+- explicitly build with `--turbopack`
+- migrate your Webpack configuration to Turbopack
+- or [opt out of Turbopack](#opting-out-of-turbopack) using `--webpack`
+
+> **Good to know**: If you see failing builds because a `webpack` configuration was found, but you don't define one yourself, it is likely that a plugin is adding a `webpack` option
+
+### Opting out of Turbopack
+
+If you need to continue using Webpack, you can opt out with the `--webpack` flag. For example, to use Turbopack in development but Webpack for production builds:
+
+```json filename="package.json"
+{
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build --webpack",
+    "start": "next start"
+  }
+}
+```
+
+We recommend using Turbopack for development and production. Submit a comment to this [thread](https://github.com/vercel/next.js/discussions/77721), if you are still unable to switch to Turbopack.
+
+### Turbopack Configuration Location
+
+The `experimental.turbopack` configuration is out of experimental.
+
+```ts filename="next.config.ts"
+import type { NextConfig } from 'next'
+
+// Next.js 15 - experimental.turbopack
+const nextConfig: NextConfig = {
+  experimental: {
+    turbopack: {
+      // options
+    },
+  },
+}
+
+export default nextConfig
+```
+
+It is now moved to be top-level option `turbopack`:
+
+```ts filename="next.config.ts"
+import type { NextConfig } from 'next'
+
+// Next.js 16 - turbopack at the top level of nextConfig
+const nextConfig: NextConfig = {
+  turbopack: {
+    // options
+  },
+}
+
+export default nextConfig
+```
+
+Make sure to review the `Turbopack` configuration [options](/docs/app/api-reference/config/next-config-js/turbopack). **Next.js 16** introduces various improvements and new options, for example:
+
+- [Advanced Webpack loader conditions](/docs/app/api-reference/config/next-config-js/turbopack#advanced-webpack-loader-conditions)
+- [debugIds](/docs/app/api-reference/config/next-config-js/turbopack#debug-ids)
+
+### Resolve alias fallback
+
+In some projects client-side code, may import files containing Node.js native modules. This causes `Module not found: Can't resolve 'fs'` type of errors.
```

Comment:
```suggestion
In some projects, client-side code may import files containing Node.js native modules. This will cause `Module not found: Can't resolve 'fs'` type of errors.
```